pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -2,21 +2,22 @@
 , lib
 , zlib
 , useCudatoolkitRunfile ? false
+, config
 , cudaVersion
 , cudaMajorVersion
 , cudatoolkit # if cuda>=11: only used for .cc
 , libcublas ? null # cuda <11 doesn't ship redist packages
 , autoPatchelfHook
 , autoAddOpenGLRunpathHook
 , fetchurl
-, # The distributed version of CUDNN includes both dynamically liked .so files,
+, # The distributed version of CUDNN includes both dynamically linked .so files,
   # as well as statically linked .a files.  However, CUDNN is quite large
   # (multiple gigabytes), so you can save some space in your nix store by
   # removing the statically linked libraries if you are not using them.
   #
   # Setting this to true removes the statically linked .a files.
   # Setting this to false keeps these statically linked .a files.
-  removeStatic ? false
+  removeStatic ? config.cudaRemoveStatic or false
```

Comment:
Is this the only place that `config.cudaRemoveStatic` is referenced? If so, why not control this setting with an overlay for the cudnn package?

Reply:
This is presently the only place where the feature is available, but I do not expect it to stay this way.  It is not unique to CuDNN.  There are also static libraries built in other parts of CUDA (e.g., cudatoolkit) and it is my hope that maintainers of those packages adopt a common approach flag alongside the other ongoing cleanups.

Reply:
Hmm, I see where you're coming from here. I'm not necessarily opposed to providing that flag eventually, but I'd like to keep that refactor for a separate PR once it actually becomes relevant in multiple places. Otherwise we risk ending up in this limbo situation where we only have a single user of this flag and it's forgotten why it's there and what it was intended for. Best to keep scope of each PR as small as possible.

Reply:
Makes sense to move to another PR.  I've removed it from this.